### PR TITLE
fix: add log when failed to scan confirmation

### LINKED ISSUE #704: manifest and blob deletion might be silently cancelled without logging
### What happened in your environment?

`oras manifest delete` and `oras blob delete` can be cancelled without logging if the stdin stream is closed before scanning

### What did you expect to happen?

Should output `Operation cancelled` if deletion is not confirmed via stdin or flag.

### How can we reproduce it?

Save below snippet to test.go, Replace <your-image> with a image name  and run it via `go run test.go`

```golang
package main

import (
	"fmt"
	"io"
	"os"
	"os/exec"
	"strings"
)

func main() {
	var cmd *exec.Cmd

	fmt.Println("Should exit with no logging")
	stdin := io.NopCloser(strings.NewReader(""))
	cmd = exec.Command("oras", "manifest", "delete", "<your-image>")
	cmd.Stderr = os.Stderr
	cmd.Stdout = os.Stdout
	cmd.Stdin = stdin
	stdin.Close()
	cmd.Start()
	cmd.Wait()
	fmt.Println()

	fmt.Println("Should exit with error logging")
	stdin = io.NopCloser(strings.NewReader("n"))
	cmd = exec.Command("oras", "manifest", "delete", "<your-image>")
	cmd.Stderr = os.Stderr
	cmd.Stdout = os.Stdout
	cmd.Stdin = stdin
	defer stdin.Close()
	cmd.Start()
	cmd.Wait()
	fmt.Println()
}

```

### What is the version of your ORAS CLI?

0.16.0

### What is your OS environment?

 Ubuntu 20.04

### Are you willing to submit PRs to fix it?

- [X] Yes, I am willing to fix it.

### PR DESCRIPTION
**Pull request by qweeah**
Resolves #704 

Signed-off-by: Billy Zha <jinzha1@microsoft.com>

### CODE REVIEW

File: cmd/oras/internal/option/confirmation.go [hunk 1]
```diff
@@ -45,7 +45,12 @@ func (opts *Confirmation) AskForConfirmation(r io.Reader, prompt string) (bool,
...
 	if ok := scanner.Scan(); !ok {
-		return false, scanner.Err()
+		err := scanner.Err()
+		if err == nil {
+			// EOF
+			fmt.Println("Operation cancelled.")
+		}
+		return false, err
 	}
 	response := scanner.Text()
```
Comment (shizhMSFT) [inline cmd/oras/internal/option/confirmation.go:55]: How about following the examples in the doc of [bufio.Scanner](https://pkg.go.dev/bufio#Scanner)?
```go
var response string
scanner := bufio.NewScanner(r)
if ok := scanner.Scan(); ok {
	response := scanner.Text()
}
if err := scanner.Err(); err != nil {
	return false, err
}
```

Comment (qweeah) [reply cmd/oras/internal/option/confirmation.go:55]: Let me check if scanning EOF breaks `strings.ToLower(response)`. If not, this is is definitely better.

Comment (qweeah) [reply cmd/oras/internal/option/confirmation.go]: updated

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**